broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManager.java

```diff
@@ -0,0 +1,142 @@
+  private final ManagedPartitionGroup partitionGroup;
+
+  protected PartitionManager(
+      final BrokerCfg configuration,
+      final ReceivableSnapshotStoreFactory snapshotStoreFactory,
+      final ClusterMembershipService membershipService,
+      final ClusterCommunicationService communicationService) {
+
+    partitionGroup = buildRaftPartitionGroup(configuration, snapshotStoreFactory);
+
+    partitions = buildPartitionService(membershipService, communicationService);
+  }
+
+  // TODO remove static modifier
+  public static RaftPartitionGroup buildRaftPartitionGroup(
+      final BrokerCfg configuration, final ReceivableSnapshotStoreFactory snapshotStoreFactory) {
+
+    final DataCfg dataConfiguration = configuration.getData();
+    final String rootDirectory = dataConfiguration.getDirectory();
+    try {
+      FileUtil.ensureDirectoryExists(new File(rootDirectory).toPath());
+    } catch (final IOException e) {
+      throw new UncheckedIOException("Failed to create data directory", e);
+    }
+
+    final File raftDirectory = new File(rootDirectory, AtomixFactory.GROUP_NAME);
+    IoUtil.ensureDirectoryExists(raftDirectory, "Raft data directory");
+
+    final ClusterCfg clusterCfg = configuration.getCluster();
+    final var experimentalCfg = configuration.getExperimental();
+    final DataCfg dataCfg = configuration.getData();
+    final NetworkCfg networkCfg = configuration.getNetwork();
+
+    final Builder partitionGroupBuilder =
+        RaftPartitionGroup.builder(AtomixFactory.GROUP_NAME)
+            .withNumPartitions(clusterCfg.getPartitionsCount())
+            .withPartitionSize(clusterCfg.getReplicationFactor())
+            .withMembers(getRaftGroupMembers(clusterCfg))
+            .withDataDirectory(raftDirectory)
+            .withSnapshotStoreFactory(snapshotStoreFactory)
+            .withMaxAppendBatchSize((int) experimentalCfg.getMaxAppendBatchSizeInBytes())
+            .withMaxAppendsPerFollower(experimentalCfg.getMaxAppendsPerFollower())
+            .withEntryValidator(new ZeebeEntryValidator())
+            .withFlushExplicitly(!experimentalCfg.isDisableExplicitRaftFlush())
+            .withFreeDiskSpace(dataCfg.getFreeDiskSpaceReplicationWatermark())
+            .withJournalIndexDensity(dataCfg.getLogIndexDensity())
+            .withPriorityElection(experimentalCfg.isEnablePriorityElection());
+
+    final int maxMessageSize = (int) networkCfg.getMaxMessageSizeInBytes();
+
+    final var segmentSize = dataCfg.getLogSegmentSizeInBytes();
+    if (segmentSize < maxMessageSize) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Expected the raft segment size greater than the max message size of %s, but was %s.",
+              maxMessageSize, segmentSize));
+    }
+
+    partitionGroupBuilder.withSegmentSize(segmentSize);
+
+    return partitionGroupBuilder.build();
+  }
+
+  public synchronized CompletableFuture<Void> start() {
+    if (closeFuture != null) {
```

Comment:
:wrench: (If we do make this an actor, then you can ignore this suggestion) I assume we're using synchronized here to ensure stop/start are mutually exclusive. I would propose to use a specific lock instead of the object's, as anybody adding more `synchronized` methods/blocks in the future might be blocking these calls unintentionally.

Reply:
It's a good point. I will ignore it for now, though. I want to wait and see how this class evolves, and then revisit synchronization again.